tests/test_ac_harness_auto_drive.py

```
@@ -0,0 +1,239 @@
+        )
+    )
+    assert report.sequence_ok is True
+    assert report.ok is False  # drove=False must veto success
+    assert report.drive is not None and report.drive.sim_dead is True
+
+
+def test_tap_exception_marks_pipeline_stage_and_still_tears_down():
+    record: dict = {}
+    ctrl = FakeController()
+
+    async def _boom(url, *, seconds, wait_for_lap):  # noqa: ANN001
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_boom`

Add return type annotation: `Never`

(ANN202)